docker/worker/Dockerfile

```diff
@@ -18,13 +18,21 @@ ENV DEBIAN_FRONTEND noninteractive
 RUN apt-get update && apt-get upgrade -y && \
     apt-get install -y \
         apt-transport-https \
+        build-essential \
         ca-certificates \
         curl \
         git \
         gnupg-agent \
-        python3-pip \
+        zlib1g-dev \
+        libffi-dev \
+        libssl-dev \
         software-properties-common
 
+RUN curl -sS https://www.python.org/ftp/python/3.11.0/Python-3.11.0.tgz | tar -C /tmp -xzv && \
+    cd /tmp/Python-3.11.0 && \
+    ./configure --enable-optimizations --with-lto --enable-loadable-sqlite-extensions && make && make install && \
```

Comment:
One consideration with using `make install` is that it may potentially cause incompatibilities with the system/apt-packaged python (if it's installed indirectly through some other package?). `altinstall` is safer because it leaves the apt python intact.

Reply:
Python3.8 is installed through the `software-properties-common` package. 

It was using `altinstall` which skips overriding the default python binary, which causes problem with pipenv (I think it was getting confused on which python version to use). Switching to `install` overrides the default python3 executable in `/usr/bin`, which seems to fix the compile issues.

Reply:
OK, hopefully there shouldn't be any backwards incompatibility issues from this.

Reply:
Honestly, this feels almost as messy as upgrading the base Docker image at this point. I'm assuming that `make altinstall` installs into `/usr/local/bin`? Can we just ensure that all invocations of `pipenv` are (for now) using `/usr/local/bin` (or better yet a central variable we can flip down the track when we _do_ upgrade the base image to Ubuntu 22.04 and can revert back to the system Python? To this end, are we going to need to adjust any shebang lines to use a different Python if we do go down the path of `make altinstall`?

I feel like we're still going to have to "test in Production" this Python change, which if we _don't_ do a `make altinstall` may result in unknown breakage of other parts of the base image.